src/MessageProtocol/IMessageProtocol.h

```diff
@@ -153,6 +153,16 @@ class IMessageProtocol
         return littleEndian;
     }
 
+    QByteArray toLittleEndianFromInt32(quint32 num)
+    {
+        QByteArray littleEndian;
+        littleEndian.append(static_cast<char>(num&0xFF));
+        littleEndian.append(static_cast<char>((num&0xFF00)>>8));
+        littleEndian.append(static_cast<char>((num&0xFF0000)>>16));
+        littleEndian.append(static_cast<char>((num&0xFF000000)>>24));
```

Comment:
there's no MSb expansion here?
IIRC it's always better to put a '0' before the FF.